locksmith/src/operations/eventCollectionOperations.ts

```diff
@@ -0,0 +1,223 @@
+          attributes: [],
+        },
+      },
+    ],
+  })
+
+  if (!eventCollection) {
+    throw new Error('Event collection not found')
+  }
+
+  return eventCollection
+}
+
+/**
+ * Updates an existing event collection.
+ * It checks if the event collection exists and if the user is authorized
+ * to make updates. If authorized, it updates the collection with the provided data.
+ *
+ * @param slug - The unique identifier (slug) of the event collection to update.
+ * @param parsedBody - The parsed body containing updated event collection details.
+ * @param userAddress - The wallet address of the user attempting to update the collection.
+ * @returns A promise that resolves to the updated event collection object.
+ * @throws An error if the collection is not found or the user is not authorized.
+ */
+export const updateEventCollectionOperation = async (
+  slug: string,
+  parsedBody: z.infer<typeof EventCollectionBody>,
+  userAddress: string
+) => {
+  const eventCollection = await EventCollection.findByPk(slug)
+  if (!eventCollection) {
+    throw new Error('Event collection not found')
+  }
+
+  if (!eventCollection.managerAddresses.includes(userAddress)) {
+    throw new Error('Not authorized to update this collection')
+  }
+
+  await eventCollection.update(parsedBody)
+  return eventCollection
+}
+
+/**
+ * Adds an event to a specified event collection.
+ * It checks for the existence of both the collection and the event
+ * before creating an association between them. If the user is a manager,
+ * they can approve the association if it was previously unapproved.
+ *
+ * @param collectionSlug - The unique identifier (slug) of the event collection
+ * @param eventSlug - The unique identifier (slug) of the event
+ * @param userAddress - The wallet address of the user attempting to add the event.
+ * @returns A promise that resolves to the event collection association object,
+ *          which includes details about the event and its approval status.
+ * @throws An error if the specified collection or event cannot be found.
+ */
+export const addEventToCollectionOperation = async (
+  collectionSlug: string,
+  eventSlug: string,
+  userAddress: string
+): Promise<{ association: EventCollectionAssociation; status: string }> => {
+  const collection = await EventCollection.findByPk(collectionSlug)
+  if (!collection) {
+    throw new Error('Collection not found')
+  }
+
+  const event = await EventData.findOne({ where: { slug: eventSlug } })
+  if (!event) {
+    throw new Error('Event not found')
+  }
+
+  const isManager = collection.managerAddresses.includes(userAddress)
+
+  const [association, created] = await EventCollectionAssociation.findOrCreate({
+    where: {
+      eventSlug: event.slug,
+      collectionSlug: collection.slug,
+    },
+    defaults: {
+      eventSlug: event.slug,
+      collectionSlug: collection.slug,
+      isApproved: isManager,
+    },
+  })
+
+  if (!created && !association.isApproved && isManager) {
```

Comment:
ok so I see this is the operation we will use to `approve` an event into a collection?